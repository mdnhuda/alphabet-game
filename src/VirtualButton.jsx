import React from 'react';
import PropTypes from 'prop-types';

const VirtualButton = ({name, value, className, clickHandler}) => (
    <button className={className} name={name} onClick={clickHandler}>{value}</button>
);

VirtualButton.propTypes = {
    value: PropTypes.string.isRequired
};

VirtualButton.defaultProps = {
    name: "",
    className: "",
    clickHandler: () => {}
};
export default VirtualButton;