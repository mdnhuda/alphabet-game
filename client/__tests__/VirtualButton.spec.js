import React from 'react';
import VirtualButton from '../app/VirtualButton';
import renderer from 'react-test-renderer';

describe('VirtualButton', () => {
    const props = {
        name: "testButton",
        value: "Test",
        className: "testClass",
        clickHandler: (e) => {}
    };

    it('renders correctly', () => {
        const tree = renderer.create(<VirtualButton {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
