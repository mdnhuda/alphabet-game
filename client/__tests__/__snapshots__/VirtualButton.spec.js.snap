// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualButton renders correctly 1`] = `
<button
  className="testClass"
  name="testButton"
  onClick={[Function]}
>
  Test
</button>
`;
